### PR TITLE
[DataStorage] API for Multiple Readings

### LINKED ISSUE #286: [Data Storage] Provide an API in Home Edge to store device generated data
Existing Problem:
Currently if any developer (using Rpi/Sensor) want to store device generated data using Home Edge, it is not possible to do. The developer has to again use EdgeX API/SDK to store the data. This makes the developer to know the working of both Home Edge and EdgeX. Home Edge should provide seamless discovery, orchestration and data sharing mechanism. For developer Home Edge should be the only point of consideration and need not bother on other solutions used internally. 

Solution:
1. Home Edge can provide an API ( POST) which can be used by developers to store the device data. The API should provide methods to store int, float, string, binary(image) data. This API can internally leverage on EdgeX to store the data.
2. We should also have an APi to get the stored values of a particular device .

### PR DESCRIPTION
**Pull request by nitu-s-gupta**
Added an API for fetching multiple readings from a device at once.

Usage Example
curl --location --request GET http://localhost:49986/api/v1/device/{DeviceName}/resource/{ResourceName}/{ReadingCount}

Signed-off-by: Nitu S Gupta <nitu.gupta@samsung.com>

# Description

This is a GET API used to get multiple stored readings of any type stored by a device

Fixes #286 

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

To test we tried to retrieve 2 and more instances of stored data  using 
```
curl --location --request GET 'http://localhost:49986/api/v1/device/datastorage/resource/string/2'
```
The output was received as desired

``` 
[
    {
        "id": "943bbbbf-f23b-481a-91fe-7c3f3042b01f",
        "created": 1626420187818,
        "origin": 1626420187745042505,
        "device": "datastorage",
        "name": "string",
        "value": "Sunchit",
        "valueType": "String"
    },
    {
        "id": "be20c158-9f9c-4b70-ae29-e167c39d9d61",
        "created": 1626344245672,
        "origin": 1626344245671619855,
        "device": "datastorage",
        "name": "string",
        "value": "Nitu Gupta",
        "valueType": "String"
    }
]
```

**Test Configuration**:
* Firmware version: Ubuntu 20.04
* Hardware: x86-64
* Edge Orchestration Release: 1.0.0

 # Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by t25kim — APPROVED** [review]
LGTM. Thanks!

**Review by tdrozdovsky — APPROVED** [review]
LGTM

**Review by MoonkiHong — APPROVED** [review]
LGTM. 😄

File: internal/controller/storagemgr/storagedriver/storagehandler.go [hunk 1]
```diff
@@ -38,11 +38,13 @@ import (
...
-	deviceNameKey     = "deviceName"
-	resourceNameKey   = "resourceName"
-	apiResourceRoute  = clients.ApiBase + "/resource/{" + deviceNameKey + "}/{" + resourceNameKey + "}"
-	handlerContextKey = "StorageHandler"
-	configPath        = "res/configuration.toml"
+	deviceNameKey            = "deviceName"
+	resourceNameKey          = "resourceName"
+	numberOfReadings         = "readingCount"
+	apiResourceRoute         = clients.ApiBase + "/resource/{" + deviceNameKey + "}/{" + resourceNameKey + "}"
+	apiResourceRouteMultiple = clients.ApiBase + "/resource/{" + deviceNameKey + "}/{" + resourceNameKey + "}/{" + numberOfReadings + "}"
```
Comment (t25kim) [inline internal/controller/storagemgr/storagedriver/storagehandler.go:45]: How about adding `device` route before deviceNameKey like below?

```suggestion
	apiResourceRoute         = clients.ApiBase + "/device/{" + deviceNameKey + "}/resource/{" + resourceNameKey + "}"
	apiResourceRouteMultiple = clients.ApiBase + "/device/{" + deviceNameKey + "}/resource/{" + resourceNameKey + "}/{" + numberOfReadings + "}"
```

**Review by t25kim — CHANGES_REQUESTED** [review]
@nitu-s-gupta This PR should be updated because of the code conflict.
Please take a look at my comment and update the PR. Thanks!

File: internal/controller/storagemgr/storagedriver/storagehandler.go (hunk 1 above)
Comment (nitu-s-gupta) [reply internal/controller/storagemgr/storagedriver/storagehandler.go]: Thanks ! Done the change.

**Comment by nitu-s-gupta** [comment]
Merged the code, Modified the path as suggested.

**Comment by t25kim** [comment]
> Merged the code, Modified the path as suggested.

Thanks for your quick update!
Could you merge your commits into one and push it with force option (--force) so that there is no meaningless commit message?

**Comment by MoonkiHong** [comment]
@t25kim I merged the regarding commits into one using an option of squash commit. 😄

**Pull request merged**